hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/repair/om/TransactionInfoRepair.java

```diff
@@ -0,0 +1,131 @@
+import org.apache.hadoop.ozone.debug.RocksDBUtils;
+import org.apache.hadoop.ozone.repair.RDBRepair;
+import org.kohsuke.MetaInfServices;
+import org.rocksdb.ColumnFamilyDescriptor;
+import org.rocksdb.ColumnFamilyHandle;
+import org.rocksdb.RocksDBException;
+import picocli.CommandLine;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+import static org.apache.hadoop.ozone.OzoneConsts.TRANSACTION_INFO_KEY;
+import static org.apache.hadoop.ozone.om.OmMetadataManagerImpl.TRANSACTION_INFO_TABLE;
+
+
+/**
+ * Tool to update the highest term-index in transactionInfoTable.
+ */
+@CommandLine.Command(
+    name = "transaction",
+    description = "CLI to update the highest index in transactionInfoTable.",
+    mixinStandardHelpOptions = true,
+    versionProvider = HddsVersionProvider.class
+)
+@MetaInfServices(SubcommandWithParent.class)
+public class TransactionInfoRepair
+    implements Callable<Void>, SubcommandWithParent  {
+
+  @CommandLine.Spec
+  private static CommandLine.Model.CommandSpec spec;
+
+  @CommandLine.ParentCommand
+  private RDBRepair parent;
+
+  @CommandLine.Option(names = {"--highest-transaction"},
+      required = true,
+      description = "highest termIndex of transactionInfoTable")
+  private String highestTransactionInfo;
+
+
+  @Override
+  public Void call() throws Exception {
+    List<ColumnFamilyHandle> cfHandleList = new ArrayList<>();
+    List<ColumnFamilyDescriptor> cfDescList = RocksDBUtils.getColumnFamilyDescriptors(parent.getDbPath());
+
+    try (ManagedRocksDB db = ManagedRocksDB.open(parent.getDbPath(), cfDescList, cfHandleList)) {
+      ColumnFamilyHandle transactionInfoCfh = getTransactionInfoCfh(cfHandleList);
+      if (transactionInfoCfh == null) {
+        System.err.println(TRANSACTION_INFO_TABLE + " is not in a column family in DB for the given path.");
+        return null;
+      }
+      TransactionInfo transactionInfo = TransactionInfo.valueOf(highestTransactionInfo);
+
+      byte[] transactionInfoBytes = TransactionInfo.getCodec().toPersistedFormat(transactionInfo);
+      db.get()
+          .put(transactionInfoCfh, StringCodec.get().toPersistedFormat(TRANSACTION_INFO_KEY), transactionInfoBytes);
+
+      System.out.println("Highest Transaction Info is updated to : " +
+          getTransactionInfo(db, transactionInfoCfh));
+    } catch (RocksDBException exception) {
+      System.err.println("Failed to update the RocksDB for the given path: " + parent.getDbPath());
+      System.err.println(
+          "Make sure that Ozone entity (OM, SCM or DN) is not running for the give dbPath and current host.");
+      System.err.println(exception);
+    } finally {
+      IOUtils.closeQuietly(cfHandleList);
+    }
+
+    return null;
+  }
+
+
+  private ColumnFamilyHandle getTransactionInfoCfh(List<ColumnFamilyHandle> cfHandleList) throws RocksDBException {
```

Comment:
nit: This can be moved to the helper class's function to reduce duplicate code.
Same for `getTransactionInfo`. It could be T type function.

Reply:
yeah, I was actually thinking the same! Will move it to another helper class's method!

Reply:
Thanks for moving it to helper function.